### ISSUE TITLE
Show notifications for inactive (but logged-in) accounts while the app is foregrounded.

### ISSUE THREAD

**Issue by chrisbobbe** [zulip/zulip-mobile#4241]
A user on iOS reports [here](https://chat.zulip.org/#narrow/stream/48-mobile/topic/%28no.20topic%29/near/1000730) that they'd like to see these notifications.

A post from @gnprice, https://github.com/zulip/zulip-mobile/issues/3114#issuecomment-585966226, reminds me that we don't actually show any notifications on iOS when the app is foregrounded. (In fact, not only is the visible alert suppressed, but the badge count isn't updated (a latent bug because servers aren't sending badge counts)—I mention this, with a proposed fix, at https://github.com/zulip/zulip-mobile/pull/4163/files#r441233689.)

The not-showing-notifications issue described at #3114 is broader than this issue, though. Ideally, we would even show notifications for the active account, if they refer to messages in a narrow that isn't visible (or perhaps, even for messages that are in the visible narrow but currently off-screen).

I'm filing this new issue, anticipating that we'll want to get there incrementally. First, we'll want to land something like https://github.com/zulip/zulip-mobile/pull/4163, to fix our confusing iOS dependencies for notifications. But, looking at that PR (and particularly the code near the [comment](https://github.com/zulip/zulip-mobile/pull/4163/files#r441233689) I linked to above), it seems like we need to say *in the iOS-native layer* whether we want to show or suppress a notification. So, we'll want to either

- store enough data in that layer for the decision to be made there, or
- make the decision in the JS layer, and report it to the iOS-native layer somehow

We'll need to do some architecture brainstorming about this; that should probably happen on CZO.

**Comment by chrisbobbe** [comment]
Marking as blocked by https://github.com/zulip/zulip-mobile/pull/4163; I expect it to land relatively soon, though.

**Comment by gnprice** [comment]
#4163 is merged!

**Comment by gnprice** [comment]
In my testing just now, I do indeed get these notifications, on iOS just like on Android. I know we've made some changes to the relevant code in the years since this was filed, and that must have resolved this.

We still don't do the more complex task of *suppressing* notifications when they're redundant with the conversation you're looking at. But that's tracked as #3114.

**Comment by gnprice** [comment]
Specifically it looks like d4606ab26c650fae661dd52cfe41f4691ae5d374 (in #4898) was the change that fixed this, back in 2021-09.
